src/crewai/task.py

```diff
@@ -204,6 +204,7 @@ def _execute_core(
         tools: Optional[List[Any]],
     ) -> TaskOutput:
         """Run the core execution logic of the task."""
+        self.agent = agent
```

Comment:
Why do we need to do this?

Self.agent isn't referenced anywhere else in _execute_core